### PR TITLE
Add Mac VPN app signing

### PR DESCRIPTION
**Pull request by hneiva**

### CODE REVIEW

File: taskcluster/ci/config.yml [hunk 1]
```diff
@@ -82,4 +82,7 @@ release-promotion:
 private-artifact-prefix: releng/adhoc
 
 scriptworker:
-    scope-prefix: project:adhoc:releng
+    scope-prefix:
+        by-product:
+            mozillavpn: project:releng:signing
+            default: project:adhoc:releng
```
Comment (hneiva) [inline taskcluster/ci/config.yml:88]: Not sure if we want to change the scope here, or if we want to add the adhoc scope to the `script_config.yml` in iscript.

Comment (escapewindow) [reply taskcluster/ci/config.yml:88]: We're changing the mac signer's trust domain from `gecko` to `adhoc`, so its scopes will change accordingly. (We'll change the scope prefix in `script_config.yml`)

File: taskcluster/adhoc_taskgraph/transforms/signing.py [hunk 2]
```diff
@@ -54,38 +56,47 @@ def build_signing_task(config, tasks):
...
-            task["worker"]["product"] = manifest["product"]
+        # Optional keys
+        for task_key, manifest_key in (
+            ("behavior", "mac-behavior"),
+            ("entitlements-url", "mac-entitlements-url"),
+            ("loginitems-entitlements-url", "mac-loginitems-entitlements-url"),
+            (
+                "nativemessaging-entitlements-url",
+                "mac-nativemessaging-entitlements-url",
+            ),
```
Comment (escapewindow) [inline taskcluster/adhoc_taskgraph/transforms/signing.py:84]: So this works, but it also means that every time we create a mozillavpn adhoc signing PR, we'll have to populate these, even if they stay static for months/quarters/years.

I mentioned we may want to hardcode this and populate it if our product is `mozillavpn` or whatever. What do you think?

File: taskcluster/adhoc_taskgraph/worker_types.py [hunk 3]
```diff
@@ -32,13 +29,16 @@
             }
         ],
         # behavior for mac iscript
-        Optional("mac-behavior"): Any(
+        Optional("behavior"): Any(
```
Comment (escapewindow) [inline taskcluster/adhoc_taskgraph/worker_types.py:32]: Let's keep this as mac-behavior.

File: taskcluster/adhoc_taskgraph/transforms/signing.py [hunk 4]
```diff
@@ -54,38 +56,47 @@ def build_signing_task(config, tasks):
...
-            task["worker"]["mac-behavior"] = manifest["mac-behavior"]
-        if "mac-entitlements-url" in manifest:
-            task["worker"]["entitlements-url"] = manifest["mac-entitlements-url"]
-        if "mac-provisioning-profile-url" in manifest:
-            task["worker"]["provisioning-profile-url"] = manifest["mac-provisioning-profile-url"]
-        if "product" in manifest:
-            task["worker"]["product"] = manifest["product"]
+        # Optional keys
+        for task_key, manifest_key in (
+            ("behavior", "mac-behavior"),
```
Comment (escapewindow) [inline taskcluster/adhoc_taskgraph/transforms/signing.py:78]: Let's keep it as `mac-behavior` in the manifest for clarity.

File: taskcluster/adhoc_taskgraph/worker_types.py [hunk 5]
```diff
@@ -51,24 +51,28 @@ def build_scriptworker_signing_payload(config, task, task_def):
...
+        "behavior",
+        "product",
+        "entitlements-url",
+        "provisioning-profile-url",
+        "loginitems-entitlements-url",
+        "nativemessaging-entitlements-url",
+        "provisioning-profile-url",
+    ):
+        if worker.get(key):
+            task_def["payload"][key] = worker[key]
```
Comment (escapewindow) [inline taskcluster/adhoc_taskgraph/worker_types.py:65]: By keying on `mac-behavior`, we know that it's mac. I think as currently written we'll copy these things in even if it's a windows signing request?

**Review by escapewindow — CHANGES_REQUESTED** [review]

**Comment by hneiva** [comment]
@escapewindow the flow of data is pretty confusing..

`manifest file` -> 1. translates into `worker` -> 2. translates into `task`

In step 1 is where we have data validation and massaging, step 2 creates the task.
What I'm doing is moving the `mac-behavior` to  `behavior` key translation from step 2 to step 1. The manifest file will remain with `mac-behavior` for mac builds, and now step 2 can assume the data is clean, and doesn't need to worry if it's mac or not, just fill in the blanks and push the task.

**Comment by escapewindow** [comment]
I'm still a bit worried that `worker.behavior` won't be intuitive for someone hacking adhoc-signing with just the signingscript (read: non-mac-signing) perspective in mind.

File: taskcluster/adhoc_taskgraph/static_task_vars.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+STATIC_VARS_BY_PRODUCT = {
```
Comment (escapewindow) [inline taskcluster/adhoc_taskgraph/static_task_vars.py:5]: Awesome :)
The only comment I have is `STATIC_VARS_BY_PRODUCT` sounds like it'll apply for all platforms, and these are mac-specific. Maybe `MAC_STATIC_VARS_BY_PRODUCT`?

Also, do we need to adjust the iscript schema/logic at all for these?

Comment (escapewindow) [reply taskcluster/adhoc_taskgraph/static_task_vars.py:5]: (have we verified a manual `taskgraph` run to make sure mac- and non-mask tasks look right?)

**Comment by escapewindow** [comment]
```
[task 2022-01-28T20:49:29.486Z] {
[task 2022-01-28T20:49:29.487Z]   "AnyOf": [
[task 2022-01-28T20:49:29.488Z]     "queue:create-task:highest:scriptworker-prov-v1/adhoc-t-mac-signing",
[task 2022-01-28T20:49:29.489Z]     "queue:create-task:very-high:scriptworker-prov-v1/adhoc-t-mac-signing",
[task 2022-01-28T20:49:29.490Z]     "queue:create-task:high:scriptworker-prov-v1/adhoc-t-mac-signing"
[task 2022-01-28T20:49:29.491Z]   ]
[task 2022-01-28T20:49:29.491Z] }
```

We probably want to copy the first two lines of [this block](https://hg.mozilla.org/ci/ci-configuration/file/tip/grants.yml#l1914) and `s,scriptworker-k8s,scriptworker-prov-v1`. (We moved all of our scriptworkers to the `scriptworker-k8s` provisioner when we moved them to k8s. The macs aren't kubernetes, so they stayed `scriptworker-prov-v1`)

**Review by escapewindow — APPROVED** [review]
I think the only issue is the STATIC_VARS name, which is also pretty easy to follow up on. Thank you!

**Pull request merged**